src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java

```diff
@@ -170,16 +170,8 @@ private void onPartitionsRevoked(Collection<TopicPartition> partitions) {
     public boolean onEmitFailure(SignalType signalType, EmitResult result) {
         if (!isActive.get()) {
             return false;
-        }
-
-        switch (result) {
-            case FAIL_NON_SERIALIZED:
-                return true;
-            case FAIL_OVERFLOW:
-                LockSupport.parkNanos(10);
-                return true;
-            default:
-                return false;
+        } else {
+            return result == EmitResult.FAIL_NON_SERIALIZED;
```

Comment:
Retrying emit on overflow will lead to hanging, as offering to the queue and draining it is performed in one thread, so there is no way for the queue to be freed. See `UnicastProcessor#tryEmitNext`